internal/repositories/hash_storage.go

```diff
@@ -0,0 +1,76 @@
+package repositories
+
+import (
+	"database/sql"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/models"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/api"
+	"github.com/sirupsen/logrus"
+)
+
+type HashStorage struct {
+	db     *sql.DB
+	alg    string
+	logger *logrus.Logger
+}
+
+// NewHashStorage creates a new struct HashService
+func NewHashStorage(db *sql.DB, alg string, logger *logrus.Logger) *HashStorage {
+	return &HashStorage{
+		db:     db,
+		alg:    alg,
+		logger: logger,
+	}
+}
+
+// Create accesses the repository to save data to the database
```

Comment:
Comments: to save data.., to get data.. - Which data? 
It is also one more point to have defined own model here.

Reply:
I understood you